src/coreComponents/constitutive/fluid/CompressibleSinglePhaseFluid.hpp

```diff
@@ -86,10 +107,52 @@ class CompressibleSinglePhaseUpdate final : public SingleFluidBaseUpdate
                         real64 & viscosity,
                         real64 & dViscosity_dPressure ) const override
   {
-    m_densRelation.compute( pressure, density, dDensity_dPressure );
+    m_densPresRelation.compute( pressure, density, dDensity_dPressure );
     m_viscRelation.compute( pressure, viscosity, dViscosity_dPressure );
   }
 
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  virtual void compute( real64 const pressure,
+                        real64 const temperature, 
+                        real64 & density,
+                        real64 & dDensity_dPressure,
+                        real64 & dDensity_dTemperature, 
+                        real64 & viscosity,
+                        real64 & dViscosity_dPressure, 
+                        real64 & dViscosity_dTemperature, 
+                        real64 & internalEnergy, 
+                        real64 & dInternalEnergy_dPressure, 
+                        real64 & dInternalEnergy_dTemperature ) const override
+  {
+    m_viscRelation.compute( pressure, viscosity, dViscosity_dPressure );
+    dViscosity_dTemperature = 0.0; 
+
+    if ( m_isThermal )
+    {
+      real64 density_pressurePart, density_temperaturePart; 
+      real64 density_pressurePart_deriv, density_temperaturePart_deriv; 
+
+      m_densPresRelation.compute( pressure, density_pressurePart, density_pressurePart_deriv ); 
+      m_densTempRelation.compute( temperature, density_temperaturePart, density_temperaturePart_deriv ); 
+
+      density = density_pressurePart * density_temperaturePart; 
```

Comment:
This works coz you are setting the reference density for the temperature relationship to be equal to 1.0. Is there a better way of implementing this? it should, at least, be documented.

Reply:
yes, this sounds like a better solution.